packages/@aws-cdk/aws-codepipeline-actions/lib/bitbucket/source-action.ts

```diff
@@ -69,6 +69,12 @@ export interface BitBucketSourceActionProps extends codepipeline.CommonAwsAction
  * @experimental
  */
 export class BitBucketSourceAction extends Action {
+  /**
+   * The name of the property that holds the ARN of the CodeStar Connection
+   * inside of the CodePipeline Artifact's metadata.
+   */
+  public static readonly CONNECTION_ARN_PROPERTY = 'codeStarConnectionArnProperty';
```

Comment:
Keep this internal inside the module. No need to expose in public API.

Reply:
Also, it shouldn't necessarily be affiliated with BitBucket

Reply:
Also, can't we deduce `codeBuildCloneOutput` somehow? Feels like redundant piece of configuration

Reply:
> Also, it shouldn't necessarily be affiliated with BitBucket

Where would you rather have it?

Reply:
> Also, can't we deduce `codeBuildCloneOutput` somehow? Feels like redundant piece of configuration

We could look at the actions that are in the Pipeline currently, and look at the class of each of them if it's `BitBucketSourceAction`, and then check the configuration object (which is `any`, so no types there), and check if the key `OutputArtifactFormat` is `'CODEBUILD_CLONE_REF'`, and if it is, then check the key `ConnectionArn`... but that seems super ugly.

Reply:
> Keep this internal inside the module. No need to expose in public API.

Done.